src/yuzu/configuration/configure_hotkeys.h

```diff
@@ -35,13 +45,24 @@ class ConfigureHotkeys : public QWidget {
     void RetranslateUI();
 
     void Configure(QModelIndex index);
+    void ConfigureController(QModelIndex index);
     std::pair<bool, QString> IsUsedKey(QKeySequence key_sequence) const;
+    std::pair<bool, QString> IsUsedControllerKey(QString key_sequence) const;
 
     void RestoreDefaults();
     void ClearAll();
     void PopupContextMenu(const QPoint& menu_location);
+    void RestoreControllerHotkey(QModelIndex index);
+    void RestoreHotkey(QModelIndex index);
 
     std::unique_ptr<Ui::ConfigureHotkeys> ui;
 
     QStandardItemModel* model;
+
+    void SetPollingResult(const Core::HID::NpadButton button, bool cancel);
+    QString GetButtonName(const Core::HID::NpadButton button) const;
+    Core::HID::EmulatedController* controller;
+    std::unique_ptr<QTimer> timeout_timer;
+    std::unique_ptr<QTimer> poll_timer;
+    std::optional<std::function<void(const Core::HID::NpadButton, const bool)>> input_setter;
```

Comment:
```suggestion
    std::optional<std::function<void(Core::HID::NpadButton, bool)>> input_setter;
```

Note that std::function also has an operator bool overload, so you can check if there's anything stored in it before executing it without needing std::optional to do that.